paasta_tools/cli/cmds/push_to_registry.py

```diff
@@ -83,6 +99,18 @@ def paasta_push_to_registry(args):
         service = service.split('services-', 1)[1]
     validate_service_name(service, args.soa_dir)
 
+    if not args.force:
+        try:
+            if is_docker_image_already_in_registry(service, args.commit):
+                paasta_print("The docker image is already in the PaaSTA docker registry. "
+                             "I'm NOT overriding the existing image. "
+                             "Add --force to override the image in the registry if you are sure what you are doing.")
+                return 0
+        except RequestException as e:
+            paasta_print("Can not connect to the PaaSTA docker registry to verify if this image exists.\n"
+                         "%s" % str(e))
+            return 1
```

Comment:
This is the only thing I would be worried about. We could have a lot broken pipelines if the registry can't talk back in 5 seconds.